pkg/apis/seedmanagement/types_managedseed.go

```diff
@@ -0,0 +1,157 @@
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ManagedSeed represents a Shoot that is registered as Seed.
+type ManagedSeed struct {
+	metav1.TypeMeta
+	// Standard object metadata.
+	metav1.ObjectMeta
+	// Specification of the ManagedSeed.
+	Spec ManagedSeedSpec
+	// Most recently observed status of the ManagedSeed.
+	Status ManagedSeedStatus
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ManagedSeedList is a list of ManagedSeed objects.
+type ManagedSeedList struct {
+	metav1.TypeMeta
+	// Standard list object metadata.
+	metav1.ListMeta
+	// Items is the list of ManagedSeeds.
+	Items []ManagedSeed
+}
+
+// ManagedSeedSpec is the specification of a ManagedSeed.
+type ManagedSeedSpec struct {
+	// Shoot references a Shoot that should be registered as Seed.
+	Shoot Shoot
+	// SeedTemplate is a template for a Seed object, that should be used to register a given cluster as a Seed.
+	// Either SeedTemplate or Gardenlet must be specified. When Seed is specified, the ManagedSeed controller will not deploy a gardenlet into the cluster
+	// and an existing gardenlet reconciling the new Seed is required.
+	SeedTemplate *SeedTemplate
+	// Gardenlet specifies, that the ManagedSeed controller should deploy a gardenlet into the cluster
+	// with the given deployment parameters and GardenletConfiguration.
+	Gardenlet *Gardenlet
+}
+
+// Shoot identifies the Shoot that should be registered as Seed.
+type Shoot struct {
+	// Name is the name of the Shoot that will be registered as Seed.
+	Name string
+}
+
+// SeedTemplate is a template for creating a Seed object, when registering a cluster as a ManagedSeed.
+type SeedTemplate struct {
+	// Standard object metadata.
+	metav1.ObjectMeta
+	// Specification of the desired behavior of the Seed.
+	Spec gardencore.SeedSpec
+}
+
+// Gardenlet specifies gardenlet deployment parameters and the GardenletConfiguration used to configure gardenlet.
+type Gardenlet struct {
+	// Deployment specifies certain gardenlet deployment parameters, such as the number of replicas,
+	// the image, etc.
+	Deployment *GardenletDeployment
+	// Config is the GardenletConfiguration used to configure gardenlet.
+	Config *runtime.RawExtension
+	// GardenConnectionBootstrap is the mechanism that should be used for bootstrapping gardenlet connection to the Garden cluster. One of ServiceAccount, BootstrapToken.
+	// If specified, a service account or a bootstrap token will be created in the garden cluster and used to compute the bootstrap kubeconfig.
+	// If not specified, the gardenClientConnection.kubeconfig field will be used to connect to the Garden cluster.
+	GardenConnectionBootstrap *GardenConnectionBootstrap
+	// DisableMergingWithParent specifies whether the deployment parameters and GardenletConfiguration of the parent gardenlet
+	// should be merged with the specified deployment parameters and GardenletConfiguration. Defaults to false.
+	DisableMergingWithParent *bool
```

Comment:
OK, it was like this and when I proposed this change, nobody objected. My original intention was to have false as default so that the user could skip if what he wants is `false`. However, with a pointer type and defaults properly implemented, this is no longer a concern and I agree the new naming is awkward. I will rename it back to `MergeWithParent`.